mobile/library/common/common/system_helper.cc

```diff
@@ -0,0 +1,11 @@
+#include "library/common/common/system_helper.h"
+
+#include "library/common/common/default_system_helper.h"
+
+namespace Envoy {
+
+std::unique_ptr<SystemHelper> SystemHelper::instance_ = std::make_unique<DefaultSystemHelper>();
```

Comment:
I like this design!  It allows us to keep a singleton instead of re-creating this system-wide object over and over, while allowing it to be swapped out easily for tests.

Reply:
Thanks! I was hoping the make this as minimal overhead as possible!